authnz/src/main/java/io/stargate/auth/entity/AuthorizedResource.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright The Stargate Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.stargate.auth.entity;
+
+import org.immutables.value.Value;
+
+/** Identifies a resource for authorization purposes. */
+@Value.Immutable
+public interface AuthorizedResource<W> extends AuthEntityWrapper<W> {
+
+  /** Identifies the resource kind. */
+  @Value.Parameter
+  ResourceKind kind();
+
+  /**
+   * Identifies the scope of the resource according to the resource {@link #kind}.
+   *
+   * <p>For roles the scope should be {@code null}.
+   *
+   * <p>For tables and functions the scope is the keyspace name. If {@code null}, it means "all
+   * keyspaces" (in which case {@link #element()} should be {@code null} too).
+   */
+  @Value.Default
+  default String scope() {
```

Comment:
I wonder if we could create an Enum that presents all possible scopes? This way we would make the API more type-safe) or reuse because I think there is already an Enum that carries the info about `Scope`)

Reply:
`scope` is either `null` of the keyspace name... I'll rename the method to `keyspace()` for clarity.